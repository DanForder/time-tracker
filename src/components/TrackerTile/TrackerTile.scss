@use "../../styles/variables" as *;
@use "../../styles/mixins" as *;

.tracker-tile {
  max-width: 15rem;
  border-radius: $border-radius;

  &--work {
    @include bg-color($color-work);
  }

  &--play {
    @include bg-color($color-play);
  }

  &--study {
    @include bg-color($color-study);
  }

  &--exercise {
    @include bg-color($color-exercise);
  }

  &--social {
    @include bg-color($color-social);
  }

  &--self-care {
    @include bg-color($color-self-care);
  }

  .tracker-tile__background-image-wrapper {
    height: 2rem;
    position: relative;
    overflow: hidden;

    .tracker-tile__background-image {
      position: absolute;
      right: 0;
      bottom: 0;
      width: 3.75rem;
      transform: translate(-1.1rem, 25%);
    }
  }

  .tracker-tile__content {
    background-color: $color-neutral-dark;
    border-radius: $border-radius;
    display: flex;
    flex-direction: column;
    padding: 1.65rem;
    gap: 0.5rem;

    .tracker-tile__content-row {
      display: grid;
      grid-template: 1fr / 1fr 1fr;

      .tracker-tile__heading {
        font-weight: $font-weight-bold;
        text-transform: capitalize;
      }

      .tracker-tile__button {
        background-color: transparent;
        border: none;
        display: flex;
        align-items: center;
        justify-content: end;
        cursor: pointer;

        .tracker-tile__button-icon {
          fill: $color-neutral-pale;
        }

        &:hover {
          .tracker-tile__button-icon {
            fill: $color-text;
          }
        }
      }

      .tracker-tile__this-week-time {
        font-size: $font-size-large;
        font-weight: $font-weight-thin;
      }

      .tracker-tile__last-week-time {
        font-size: $font-size-small;
        font-weight: $font-weight-thin;
        color: $color-neutral-pale;
        text-transform: capitalize;
        text-align: right;
      }
    }
  }
}

@media screen and (min-width: $desktop-breakpoint) {
  .tracker-tile {
    .tracker-tile__content {
      .tracker-tile__content-row {
        &:last-child {
          grid-template: 3fr 1fr / 1fr;
          gap: 0.75rem;

          .tracker-tile__this-week-time {
            font-size: $font-size-largest;
          }

          .tracker-tile__last-week-time {
            text-align: left;
          }
        }
      }
    }
  }
}
