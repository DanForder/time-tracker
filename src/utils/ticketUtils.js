export const getCategoryImage = async (category) => {
  const iconName = formatCategoryString(category);
  const image = await import(`../assets/images/icon-${iconName}.svg`);
  return image.default;
};

export const formatCategoryString = (string) => {
  return string.toLowerCase().split(" ").join("-");
};

export const getLastTimeframeName = (timeframe) => {
  return timeframe === "daily"
    ? "Yesterday"
    : `Last ${timeframe.substring(0, timeframe.length - 2)}`;
};
