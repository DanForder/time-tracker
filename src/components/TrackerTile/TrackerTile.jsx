import classnames from "classnames";
import { ReactComponent as EllipsisIcon } from "../../assets/images/icon-ellipsis.svg";
import useCategoryImage from "../../hooks/useCategoryImage";
import { formatCategoryString } from "../../utils/ticketUtils";
import "./TrackerTile.scss";

const TrackerTile = ({ category, timeframes, lastTimeframeName }) => {
  const [backgroundImage] = useCategoryImage(category);
  const rootClassNames = classnames(
    "tracker-tile",
    `tracker-tile--${formatCategoryString(category)}`
  );

  return (
    <div className={rootClassNames}>
      <div className="tracker-tile__background-image-wrapper">
        <img
          className="tracker-tile__background-image"
          src={backgroundImage}
          alt=""
        />
      </div>
      <div className="tracker-tile__content">
        <div className="tracker-tile__content-row">
          <h2 className="tracker-tile__heading">{category}</h2>
          <button className="tracker-tile__button">
            <EllipsisIcon className="tracker-tile__button-icon" />
          </button>
        </div>
        <div className="tracker-tile__content-row">
          <p className="tracker-tile__this-week-time">
            {timeframes.current}hrs
          </p>
          <p className="tracker-tile__last-week-time">
            {lastTimeframeName} - {timeframes.previous}hrs
          </p>
        </div>
      </div>
    </div>
  );
};

export default TrackerTile;
