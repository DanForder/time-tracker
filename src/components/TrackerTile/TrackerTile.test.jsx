import { customRender } from "../../utils/testUtils";
import TrackerTile from "./TrackerTile";

it("should render the tracker tile component", () => {
  const { container } = customRender(<TrackerTile category="work" />);

  expect(container).toMatchSnapshot();
});
