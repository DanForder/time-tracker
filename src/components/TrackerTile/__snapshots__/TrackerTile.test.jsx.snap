// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the tracker tile component 1`] = `
<div>
  <div
    class="tracker-tile tracker-tile--work"
  >
    <div
      class="tracker-tile__background-image-wrapper"
    >
      <img
        alt=""
        class="tracker-tile__background-image"
      />
    </div>
    <div
      class="tracker-tile__content"
    >
      <div
        class="tracker-tile__content-row"
      >
        <h2
          class="tracker-tile__heading"
        >
          work
        </h2>
        <button
          class="tracker-tile__button"
        >
          <svg
            class="tracker-tile__button-icon"
          >
            icon-ellipsis.svg
          </svg>
        </button>
      </div>
      <div
        class="tracker-tile__content-row"
      >
        <p
          class="tracker-tile__this-week-time"
        >
          hrs
        </p>
        <p
          class="tracker-tile__last-week-time"
        >
          Last Week - 
          hrs
        </p>
      </div>
    </div>
  </div>
</div>
`;
